ui/jobs/components/nin.ts

```diff
@@ -1,123 +1,161 @@
 import EffectId from '../../../resources/effect_id';
+import TimerBar from '../../../resources/timerbar';
+import TimerBox from '../../../resources/timerbox';
 import { JobDetail } from '../../../types/event';
+import { ResourceBox } from '../bars';
+import ComboTracker from '../combo_tracker';
 import { kAbility } from '../constants';
-import { Bars } from '../jobs';
 import { computeBackgroundColorFrom } from '../utils';
 
-let resetFunc: (bars: Bars) => void;
+import { BaseComponent, ComponentInterface } from './base';
 
-export const setup = (bars: Bars): void => {
-  let tid1 = 0;
+export class NINComponent extends BaseComponent {
+  ninki: ResourceBox;
+  hutonBox: TimerBox;
+  trickAttack: TimerBox;
+  bunshin: TimerBox;
+  ninjutsu: TimerBox;
+  comboTimer: TimerBar;
+  tid1 = 0;
+  mudraTriggerCd = true;
 
-  const ninki = bars.addResourceBox({
-    classList: ['nin-color-ninki'],
-  });
-  const hutonBox = bars.addProcBox({
-    id: 'nin-procs-huton',
-    fgColor: 'nin-color-huton',
-    threshold: 20,
-  });
-  const trickAttack = bars.addProcBox({
-    id: 'nin-procs-trickattack',
-    fgColor: 'nin-color-trickattack',
-  });
-  const bunshin = bars.addProcBox({
-    id: 'nin-procs-bunshin',
-    fgColor: 'nin-color-bunshin',
-  });
-  bars.onUseAbility(kAbility.Bunshin, () => {
-    bunshin.duration = 90;
-  });
-  const ninjutsu = bars.addProcBox({
-    id: 'nin-procs-ninjutsu',
-    fgColor: 'nin-color-ninjutsu',
-  });
+  constructor(o: ComponentInterface) {
+    super(o);
 
-  let mudraTriggerCd = true;
-  // Ninjutsu's cooldown begins to countdown at the first mudra.
-  bars.onYouGainEffect(EffectId.Mudra, () => {
-    if (!mudraTriggerCd)
-      return;
+    this.ninki = this.bars.addResourceBox({
+      classList: ['nin-color-ninki'],
+    });
+    this.hutonBox = this.bars.addProcBox({
```

Comment:
Both huton and ninjutsu both appear to not work, but it appears that it was broken before your patch too.